### PR TITLE
feat(new transform): Initial `rename_fields` transform implementation

### LINKED ISSUE #377: New `rename_fields` transform
A simple transformer that renames fields. The config should be a table of k/v pairs representing the old name and new name:

```
[transforms.<transform_id>]
  type = "rename_fields"

  [transforms.<transform_id>.fields]
    old_name = "new_name"
```

### PR DESCRIPTION
**Pull request by Hoverbear**
Adds the `rename_fields` transform along with some basic docs.

I think there is some space for a more exhaustive test, eg moving around tested fields or vector slots.

There is nothing surprising here. Closes #377

### CODE REVIEW

**Comment by binarylogic** [comment]
Looks great! Would you mind adding [behavior tests](https://github.com/timberio/vector/tree/master/tests/behavior/transforms) for the nested field access? @a-rodin is working on adding these for other transforms, and it'll ensure we don't break anything in https://github.com/timberio/vector/pull/1662.

File: .meta/transforms/rename_fields.toml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+"""
+
+[transforms.rename_fields.options.fields.children."`[field-name]`"]
+type = "*"
+required = true
+examples = [
+  {"old field name" = "new field name"},
+]
+description = """\
+The name of the field to move. . Use `.` for adding nested fields.\
```
Comment (binarylogic) [inline .meta/transforms/rename_fields.toml]: Looks like an extra `.` is here.

File: website/docs/reference/transforms/rename_fields.md.erb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+<%- end -%>
+<%- if component.output.log -%>
+## Output
+
+<%= component_output(component, component.output.log, heading_depth: 3) %>
+
+<%- end -%>
+## How It Works [[sort]]
+
```
Comment (binarylogic) [inline website/docs/reference/transforms/rename_fields.md.erb]: I'd add 2 sections here:

1. Nested fields
    Just a simple section that makes it obvious nested fields are supported.

2. Name collissions
    What happens when a field is renamed to an existing field? What if the existing field is an object (nested). Etc.

Comment (Hoverbear) [reply website/docs/reference/transforms/rename_fields.md.erb]: Turns out nested fields weren't working in this yet! So I added them, and added behavioral tests to check things. :)

File: src/transforms/rename_fields.rs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+use indexmap::map::IndexMap;
+use serde::{Deserialize, Serialize};
+use snafu::Snafu;
+use string_cache::DefaultAtom as Atom;
+use toml::value::Value as TomlValue;
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct RenameFieldsConfig {
+    pub fields: IndexMap<String, TomlValue>,
```
Comment (LucioFranco) [inline src/transforms/rename_fields.rs]: Maybe, I'm missing something but why can't this just be `IndexMap<Atom, Atom>`?

Comment (Hoverbear) [reply src/transforms/rename_fields.rs]: We need to support nested fields and things like `a.b.c = 'd'` come out as `a = { b = { c = 'd' }}}`

**Comment by Hoverbear** [comment]
I have not seen this error before...

**Comment by Hoverbear** [comment]
Can someone help me understand the issue with the `make generate` here?

**Pull request merged**